src/m365/flow/commands/flow-list.ts

```diff
@@ -78,23 +83,39 @@ class FlowListCommand extends AzmgmtItemsListCommand<{ name: string, displayName
   }
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
-    const url: string = `${this.resource}providers/Microsoft.ProcessSimple${args.options.asAdmin ? '/scopes/admin' : ''}/environments/${formatting.encodeQueryParameter(args.options.environmentName)}/flows?api-version=2016-11-01`;
-
     try {
-      if (args.options.asAdmin || !args.options.sharingStatus || args.options.sharingStatus === 'ownedByMe') {
+      const {
+        environmentName,
+        asAdmin,
+        sharingStatus,
+        includeSolutions
+      } = args.options;
+
+      if (sharingStatus === 'personal') {
+        const url = this.getApiUrl(environmentName, asAdmin, includeSolutions, 'personal');
         await this.getAllItems(url, logger, true);
       }
-      else if (args.options.sharingStatus === 'personal') {
-        await this.getFilteredItems(url, logger, 'personal', true);
+      else if (sharingStatus === 'sharedWithMe') {
+        const url = this.getApiUrl(environmentName, asAdmin, includeSolutions, 'team');
+        await this.getAllItems(url, logger, true);
       }
-      else if (args.options.sharingStatus === 'sharedWithMe') {
-        await this.getFilteredItems(url, logger, 'team', true);
+      else if (sharingStatus === 'all') {
+        let url = this.getApiUrl(environmentName, asAdmin, includeSolutions, 'personal');
+        await this.getAllItems(url, logger, true);
+
+        url = this.getApiUrl(environmentName, asAdmin, includeSolutions, 'team');
+        await this.getAllItems(url, logger, false);
       }
       else {
-        await this.getFilteredItems(url, logger, 'personal', true);
-        await this.getFilteredItems(url, logger, 'team', false);
+        const url = this.getApiUrl(environmentName, asAdmin, includeSolutions);
+        await this.getAllItems(url, logger, true);
       }
 
+      // Remove duplicates
+      this.items = this.items.filter((flow, index, self) =>
+        index === self.findIndex(f => f.id === flow.id)
+      );
```

Comment:
I understand we treat this as a kind of bug fix right? if not then this would probably be a breaking change right 🤔

Reply:
It should have been included in the previous PR when we introduced te shared option. But both the contribor and me forgot about it. It was my bad to not include it into the original issue message. While implementing this I noticed again that I'm getting some flows 2-3 times.
So yeah it's definitely a bigfix since it was never intentional.